code/app/src/main/java/com/project/lordofthewings/Views/ProfilePage.java

```diff
@@ -49,7 +49,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         ImageButton backButton = findViewById(R.id.backIcon);
         backButton.setOnClickListener(v -> {
-            Intent intent = new Intent(ProfilePage.this, HomePage.class);
+            Intent intent = new Intent(ProfilePage.this, LeaderBoardPage.class);
             startActivity(intent);
             finish();
```

Comment:
functionally this code will cause activity flow errors since it will also send you back you to the leaderboard page if you are on your own profile.

this is unintended behaviour

Reply:
+1